pkg/skaffold/event/server.go

```diff
@@ -84,3 +91,37 @@ func newStatusServer(port string) (func() error, error) {
 		return l.Close()
 	}, nil
 }
+
+// getOpenPort tests the provided port for availability,
+// and if it's already in use, finds another open port.
+func getAvailablePort(port string) string {
```

Comment:
You may be able to reuse some of the port-forwarding functions here --
[getAvailablePort](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/kubernetes/port_forward.go#L274)
[portAvailable](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/kubernetes/port_forward.go#L290)

Reply:
good call. I did some refactoring that will have a slight functional change on the way we select ports when forwarding, so gonna close this PR and open that one first, then reopen this on top of those changes.